### PR TITLE
Adapt CLI to new D-Bus API

### PR DESCRIPTION
**Pull request by joseivanlopez**
## Problem

Several changes were done in the D-Bus API, and the CLI code is not adapated to such changes yet.

This is a follow-up of https://github.com/yast/d-installer/pull/209.

## Solution

Adapt CLI to the new D-Bus API.

## Testing

* Added new unit tests
* Tested manually

### CODE REVIEW

File: service/test/dinstaller/dbus/clients/manager_test.rb [hunk 1]
```diff
@@ -44,24 +48,59 @@
 
   subject { described_class.new }
 
+  describe "#Probe" do
+    # Using partial double because methods are dynamically added to the proxy object
```
[the remark this answers is not in the document]
Comment (joseivanlopez) [reply service/test/dinstaller/dbus/clients/manager_test.rb:52]: Class Foo does not implement instance method #bar, or something like that.

**Review by mvidner — CHANGES_REQUESTED** [review]
```sh
cd d-installer/service
bundle exec rspec
```

 2 tests fail because they cannot get the storage lock (as non root).

File: service/lib/dinstaller/dbus/clients/manager.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      # D-Bus client for manager service
+      class Manager < Base
+        include WithServiceStatus
+        include WithProgress
+
+        def initialize
+          super
+
+          @dbus_object = service.object("/org/opensuse/DInstaller/Manager1")
+          @dbus_object.introspect
```
[the remark this answers is not in the document]
Comment (joseivanlopez) [reply service/lib/dinstaller/dbus/clients/manager.rb:40]: As discussed on IRC, this call to `#introspect` is still needed.

[the remark this answers is not in the document]
Comment (mvidner) [reply service/lib/dinstaller/dbus/clients/manager.rb:40]: > if I omit #introspect, then methods like [DBus::Clients::Software#select_product](https://github.com/yast/d-installer/blob/master/service/lib/dinstaller/dbus/clients/software.rb#L70) are not found

**Comment by mvidner** [comment]
The change from `DBus::Service#object` to `DBus::Service#[]` has silently fixed a bug in `DInstaller::DBus::Clients::Software#provision_selected?` - previously it would return `[true]` or `[false]` which are both truthy.
Checked with irb.

**Review by mvidner — APPROVED** [review]

**Pull request merged**